### PR TITLE
RESOURCE-136 F/df ds

### PR DESCRIPTION
**Pull request by sa-progress**
### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec](https://github.com/inspec/inspec-azure/CONTRIBUTING.MD) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] `rake lint` passes
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_data_factory_datasets.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## Azure REST API version, endpoint, and HTTP client parameters
+
+This resource interacts with api versions supported by the resource provider.
+The `api_version` can be defined as a resource parameter.
+If not provided, the latest version will be used.
+For more information, refer to [`azure_generic_resource`](azure_generic_resource.md).
+
+Unless defined, `azure_cloud` global endpoint, and default values for the HTTP client will be used.
+For more information, refer to the resource pack [README](../../README.md).
+For api related info : [`Azure dataset Docs`](https://docs.microsoft.com/en-us/rest/api/securityinsights/alert-rules/list).
```
Comment (sathish-progress) [inline docs/resources/azure_data_factory_datasets.md:19]: please add datasets link

File: libraries/azure_backend.rb [hunk 2]
```diff
@@ -580,6 +580,8 @@ def create_methods(object, data)
       data.each do |key, value|
         create_method(object, key, value)
       end
+      # require 'byebug'
+      # byebug
```
Comment (sathish-progress) [inline libraries/azure_backend.rb:584]: Could you please remove this ?

File: libraries/azure_data_factory_datasets.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  def initialize(opts = {})
+    # Options should be Hash type. Otherwise Ruby will raise an error when we try to access the keys.
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.DataFactory/factories', opts)
+    opts[:required_parameters] = %i(factory_name)
+    opts[:resource_path] = [opts[:factory_name], 'datasets'].join('/')
+    # require 'byebug'
+    # byebug
```
Comment (sathish-progress) [inline libraries/azure_data_factory_datasets.rb:22]: same here

**Review by sathish-progress — APPROVED** [review]
Everything else LGTM except the small typo with link

**Review by IanMadd — APPROVED** [review]

**Comment by clintoncwolfe** [comment]
@sa-progress could you please resolve the conflicts?

**Pull request merged**